cmd/maya-apiserver/cstor-operator/cspc/operations.go

```diff
@@ -83,6 +83,13 @@ func (pc *PoolConfig) addGroupToPool(cspcPoolSpec *apis.PoolSpec, csp *apis.NewT
 			if cspcRaidGroup.Type == "" {
 				cspcRaidGroup.Type = cspcPoolSpec.PoolConfig.DefaultRaidGroupType
 			}
+
```

Comment:
instead of this, and next for loop, how about calling `algorithmconfig.AreBDsUsable(array of BD names)` which is nothing but rename of `algorithmconfig.ClaimBDsForNode(array of BD names)`?
same applies to the next chunk of code as well

Reply:
Also, lets remove multi-return value from `algorithmconfig.IsClaimedBDUsable`.. lets just return error if BD is not usable.. so, at the `return` statement which is at the end of function, lets return error saying `BD is already in use`.. this avoids multiple checks in the caller

Reply:
lets rename `algorithmconfig.IsClaimedBDUsable` to `algorithmconfig.IsBDUsable` as the BD name that is passed might be not claimed..

Reply:
This may involve changes that is out of scope of this PR.
Will consider it later.